### PR TITLE
Cu 82pcg4 css integration button

### PR DESCRIPTION
**Pull request by Hamza-A**
This branch aims to implement the Button stylized as on the wireframe everywhere we have a button on the app. To do this, we are now using Storybook which helps us create new components and their documentation. 

## Description

I had created a new `<Button />` display component and its story by the way. This component includes 4 variants `Primary` `SECONDARY` `DANGER` and `GHOST`. They have been implemented in the whole application.
## How Has This Been Tested?

I've replaced old buttons by new ones in test files.

## Screenshots

image
image
image
image
image
image

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- :green_circle: My code follows the code style of this project.
- :red_circle: My change requires a change to the documentation.
- :red_circle: I have updated the documentation accordingly.
- :green_circle: I have read the [**CONTRIBUTING**][contributing_file] document.
- :green_circle: I have added tests to cover my changes.
- :green_circle: All new and existing tests passed.

[contributing_file]: https://github.com/fewlinesco/connect/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: src/@types/ButtonVariant.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export enum ButtonVariant {
+  PRIMARY = "PRIMARY",
+  SECONDARY = "SECONDARY",
+  DANGER = "DANGER",
+  GHOST = "GHOST",
+}
```
Comment (Fuzznimp) [inline src/@types/ButtonVariant.ts:6]: Hm.. I'm thorned between putting this in the button file, or in `/@types/components/ButtonVarian`.

Do you have a preference?

File: src/components/display/fewlines/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import { ButtonVariant } from "../../../@types/ButtonVariant";
+
+interface ButtonProps {
+  variant?: ButtonVariant;
+}
+
+export const Button = styled.button<ButtonProps>`
+  height: 4rem;
+  border-radius: 0.2rem;
```
Comment (Fuzznimp) [inline src/components/display/fewlines/Button.tsx:11]: Use the first value of the theme's `radii` list.

File: src/components/display/fewlines/IdentityValidationForm.tsx [hunk 3]
```diff
@@ -21,43 +23,53 @@ const IdentityValidationForm: React.FC<{ type: IdentityTypes }> = ({
...
+    margin: 0 0 ${({ theme }) => theme.spaces.component.xxs} 0;
+  }
+
+  .resend-button {
+    margin: ${({ theme }) => theme.spaces.component.xxs} 0;
+  }
+
+  button {
+    width: 100%;
   }
```
Comment (Fuzznimp) [inline src/components/display/fewlines/IdentityValidationForm.tsx:68]: I'd put it before the classes, from the most generic to the most precise:
```css
 button {
    width: 100%;
  }

  .send-button {
    margin: ${({ theme }) => theme.spaces.component.xxs} 0;
  }

...
```

File: src/components/display/fewlines/UpdateIdentityForm.tsx [hunk 4]
```diff
@@ -27,15 +29,38 @@ export const UpdateIdentityForm: React.FC<{
...
+  max-width: 95%;
+  margin: 0 auto;
+
+  .send-button {
+    margin: ${({ theme }) => theme.spaces.component.xxs} 0;
+  }
+
+  button {
+    width: 100%;
+  }
```
Comment (Fuzznimp) [inline src/components/display/fewlines/UpdateIdentityForm.tsx:60]: Same, put it before `.send-button`.

**Review by Fuzznimp — CHANGES_REQUESTED** [review]

File: src/@types/ButtonVariant.ts (hunk 1 above)
Comment (Hamza-A) [reply src/@types/ButtonVariant.ts:6]: Actually I was following the rule of moving a type to its own file if it's used in several files, for me its suitable like this, the only advantage of moving it back to the component file is that we could have a direct overview of the variant's list. Do you want me to fix it?

File: src/components/display/fewlines/Button.tsx (hunk 2 above)
Comment (Hamza-A) [reply src/components/display/fewlines/Button.tsx]: OK fixed 7e41855

File: src/components/display/fewlines/IdentityValidationForm.tsx (hunk 3 above)
Comment (Hamza-A) [reply src/components/display/fewlines/IdentityValidationForm.tsx]: OK fixed 7e41855

File: src/components/display/fewlines/UpdateIdentityForm.tsx (hunk 4 above)
Comment (Hamza-A) [reply src/components/display/fewlines/UpdateIdentityForm.tsx:56]: OK fixed 7e41855

File: src/@types/ButtonVariant.ts (hunk 1 above)
Comment (Fuzznimp) [reply src/@types/ButtonVariant.ts:6]: > Actually I was following the rule of moving a type to its own file if it's used in several files

Yes, and I'm glad you did it :). It's just that `@/types` is gonna be massive if we put the components' types at the root. Let's do `@types/components/*` for now :).

Comment (Hamza-A) [reply src/@types/ButtonVariant.ts]: OK fixed 87d4ee7

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**